testsupport/tiers/checks.go

```diff
@@ -1070,16 +1071,23 @@ func clusterObjectsChecks(checkCreator ...clusterObjectsCheckCreator) []clusterO
 func idlers(timeoutSeconds int, namespaceTypes ...string) clusterObjectsCheckCreator {
 	return func() clusterObjectsCheck {
 		return func(t *testing.T, memberAwait *wait.MemberAwaitility, userName, tierLabel string) {
+			idlerWaitCriterion := []wait.IdlerWaitCriterion{
+				wait.IdlerHasTier(tierLabel),
+				wait.IdlerHasTimeoutSeconds(timeoutSeconds),
+			}
+			// cast generic labels wait criterion into idler wait criterion
+			for expectedKey, expectedValue := range toolchainLabels(userName) {
+				idlerWaitCriterion = append(idlerWaitCriterion, wait.IdlerHasLabel(expectedKey, expectedValue))
+			}
```

Comment:
We could maybe start using generics for the wait criteria so we could easily reuse them in all functions. 
but that's for later - not now :wink:

Reply:
Yeah I had the same feeling while writing this code :)